### PR TITLE
Create task for Status Check

### PR DESCRIPTION
**Pull request by aballiet**
Fixes https://github.com/kestra-io/plugin-dbt/issues/62

### CODE REVIEW

**Comment by aballiet** [comment]
A quick note regarding reproducibility locally, for dbt plugin, when setting parseRunResults(true), run is failing because : https://github.com/kestra-io/plugin-dbt/blob/65ac06fdd112e547be673e9a45ba1fcbf0298ac7/src/main/java/io/kestra/plugin/dbt/ResultParser.java#L90

image

File: src/main/java/io/kestra/plugin/dbt/cloud/CheckStatus.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+import static io.kestra.core.utils.Rethrow.throwSupplier;
+import static java.lang.Math.max;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+        title = "Check Job status of a running job"
```
Comment (brian-mulier-p) [inline src/main/java/io/kestra/plugin/dbt/cloud/CheckStatus.java:42]:
```suggestion
        title = "Check job status of a running job"
```

File: src/main/java/io/kestra/plugin/dbt/cloud/CheckStatus.java [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+    @Schema(
+            title = "The run id to check status for"
+    )
+    @PluginProperty(dynamic = true)
+    Integer runId;
+
+
+    @Schema(
+            title = "Specify frequency for job state check API calls"
```
Comment (brian-mulier-p) [inline src/main/java/io/kestra/plugin/dbt/cloud/CheckStatus.java:70]:
```suggestion
            title = "Specify frequency at which the job status will be checked"
```

File: src/main/java/io/kestra/plugin/dbt/cloud/TriggerRun.java [hunk 3]
```diff
@@ -58,11 +58,6 @@
     }
 )
 public class TriggerRun extends AbstractDbtCloud implements RunnableTask<TriggerRun.Output> {
```
Comment (brian-mulier-p) [inline src/main/java/io/kestra/plugin/dbt/cloud/TriggerRun.java:60]: loggedSteps & loggedStatus aren't used anymore there am I wrong ?

**Review by brian-mulier-p — CHANGES_REQUESTED** [review]
Thank's a lot ! :+1:

File: src/main/java/io/kestra/plugin/dbt/cloud/TriggerRun.java (hunk 3 above)
Comment (aballiet) [reply src/main/java/io/kestra/plugin/dbt/cloud/TriggerRun.java:60]: Correct !

Comment (aballiet) [reply src/main/java/io/kestra/plugin/dbt/cloud/TriggerRun.java:47]: Fixed 👍

File: src/main/java/io/kestra/plugin/dbt/cloud/CheckStatus.java [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+import static io.kestra.core.utils.Rethrow.throwSupplier;
+import static java.lang.Math.max;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+        title = "Check job status of a running job"
```
Comment (anna-geller) [inline src/main/java/io/kestra/plugin/dbt/cloud/CheckStatus.java:42]:
```suggestion
        title = "Check the status of a dbt Cloud job"
```

File: src/main/java/io/kestra/plugin/dbt/cloud/CheckStatus.java [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+)
+public class CheckStatus extends AbstractDbtCloud implements RunnableTask<CheckStatus.Output> {
+    private static final List<JobStatusHumanizedEnum> ENDED_STATUS = List.of(
+            JobStatusHumanizedEnum.ERROR,
+            JobStatusHumanizedEnum.CANCELLED,
+            JobStatusHumanizedEnum.SUCCESS
+    );
+
+    @Schema(
+            title = "The run id to check status for"
```
Comment (anna-geller) [inline src/main/java/io/kestra/plugin/dbt/cloud/CheckStatus.java:63]:
```suggestion
            title = "The job run id to check the status for"
```

File: src/main/java/io/kestra/plugin/dbt/cloud/CheckStatus.java [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+    )
+    @PluginProperty(dynamic = true)
+    Integer runId;
+
+
+    @Schema(
+            title = "Specify frequency at which the job status will be checked"
```
Comment (anna-geller) [inline src/main/java/io/kestra/plugin/dbt/cloud/CheckStatus.java:70]:
```suggestion
            title = "Specify how often the task should poll for the job status"
```

File: src/main/java/io/kestra/plugin/dbt/cloud/CheckStatus.java [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+    )
+    @PluginProperty(dynamic = false)
+    @Builder.Default
+    Duration pollFrequency = Duration.ofSeconds(5);
+
+    @Schema(
+            title = "The max total wait duration"
```
Comment (anna-geller) [inline src/main/java/io/kestra/plugin/dbt/cloud/CheckStatus.java:77]:
```suggestion
            title = "The maximum duration the task should poll for the job completion"
```

File: src/main/java/io/kestra/plugin/dbt/cloud/CheckStatus.java [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+    )
+    @PluginProperty(dynamic = false)
+    @Builder.Default
+    Duration maxDuration = Duration.ofMinutes(60);
+
+    @Builder.Default
+    @Schema(
+            title = "Parse run result",
+            description = "Parsing run result to display duration of each task inside dbt"
```
Comment (anna-geller) [inline src/main/java/io/kestra/plugin/dbt/cloud/CheckStatus.java:86]:
```suggestion
            description = "Whether to parse the run result to display the duration of each dbt node in the Gantt view"
```

File: src/main/java/io/kestra/plugin/dbt/cloud/CheckStatus.java [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+        Files.writeString(tempFile, artifact, StandardOpenOption.TRUNCATE_EXISTING);
+
+        return tempFile;
+    }
+
+    @Builder
+    @Getter
+    public static class Output implements io.kestra.core.models.tasks.Output {
+        @Schema(
+                title = "URI of a run results"
```
Comment (anna-geller) [inline src/main/java/io/kestra/plugin/dbt/cloud/CheckStatus.java:242]:
```suggestion
                title = "URI of the run result"
```

**Review by anna-geller — APPROVED** [review]
functionality-wise, LGTM

Thanks for adding this 👍

**Review by brian-mulier-p — APPROVED** [review]
LGTM

**Pull request merged**